packages/@react-aria/utils/src/useId.ts

```diff
@@ -44,3 +44,15 @@ export function mergeIds(a: string, b: string): string {
 
   return b;
 }
+
+export function useSlotId(): string {
+  let [id, setId] = useState(useId());
+  useLayoutEffect(() => {
+    let setCurr = map.get(id);
+    if (setCurr && !document.getElementById(id)) {
+      setId(null);
```

Comment:
So if the `id` does not find a home in the DOM, we set it back to null?

Reply:
yes